### PR TITLE
Updated dependencies

### PR DESCRIPTION
**Pull request by danielceinos**
### PR's key points
 Updated dependencies:
- Kotlin from `1.3.31` to `1.4.10`
- Build tools from `3.4.1` to `4.0.1`
- App compact from `1.0.2` to `1.2.0`
- Google play core from `1.6.1` to `1.8.0` **Breaking changes**

Also added plugin to version lib from git tags
 
### Definition of Done
- [x] Tests pass
- [x] Works with Proguard
- [x] There is no outcommented or debug code left (if there is a good reason, use a // FIXME: comment or // TODO: comment and create a follow-up issue)

### CODE REVIEW

File: lib/src/main/java/com/bq/appupdateshelper/FakeAppUpdatesHelper.java [hunk 1]
```diff
@@ -63,26 +63,6 @@ public void setUpdateAvailable(int availableVersionCode) {
...
-     * In case the update is only partially allowed, which means that not all in-app update types are allowed by policy,
-     * the type of update which is exclusively allowed can be set here.
-     * <p>
-     * Also sets the error code to {@link InstallErrorCode.NO_ERROR_PARTIALLY_ALLOWED} to enable the partial error behaviour.
-     *
-     * @param allowedUpdate one of {@link AppUpdateType.FLEXIBLE} or {@link AppUpdateType.IMMEDIATE}, or null
-     */
-    public void setPartiallyAllowedUpdateType(@Nullable @AppUpdateType Integer allowedUpdate) {
-        fakeAppUpdateManager.setPartiallyAllowedUpdateType(allowedUpdate);
-    }
```
Comment (danielceinos) [inline lib/src/main/java/com/bq/appupdateshelper/FakeAppUpdatesHelper.java:84]: This methods disapear on last version :/

Comment (danielceinos) [reply lib/src/main/java/com/bq/appupdateshelper/FakeAppUpdatesHelper.java:84]: https://developer.android.com/reference/com/google/android/play/core/release-notes

There's no info about it...

File: lib/build.gradle [hunk 2]
```diff
@@ -1,14 +1,20 @@
 apply plugin: 'com.android.library'
+apply plugin: "com.gladed.androidgitversion"
 
-android {
-    compileSdkVersion 29
+androidGitVersion {
+    codeFormat "MMNNPP"
+    format '%tag%'
+    tagPattern(/^[0-9]+.*/) // Tag names should follow the pattern vMM.NN.PP
```
Comment (adriangl) [inline lib/build.gradle]:
```suggestion
    tagPattern(/^[0-9]+.*/) // Tag names should follow the pattern MM.NN.PP
```

File: lib/src/main/java/com/bq/appupdateshelper/FakeAppUpdatesHelper.java (hunk 1 above)
Comment (adriangl) [reply lib/src/main/java/com/bq/appupdateshelper/FakeAppUpdatesHelper.java:84]: They may have removed it without warning 🤔?

**Review by adriangl — CHANGES_REQUESTED** [review]
Thanks for the contribution @danielceinos! 
I'll take a deeper look into it on the weekend, but for now just add the suggestion I did and fix the conflicts with the base branch

**Comment by danielceinos** [comment]
@adriangl how we could test that the new version still works? using internal testing works good?

File: lib/src/main/java/com/bq/appupdateshelper/FakeAppUpdatesHelper.java (hunk 1 above)
Comment (danielceinos) [reply lib/src/main/java/com/bq/appupdateshelper/FakeAppUpdatesHelper.java:84]: Weird..

What was the function of that methods?

Comment (adriangl) [reply lib/src/main/java/com/bq/appupdateshelper/FakeAppUpdatesHelper.java:84]: Seems like they did something with partial updates in `v1.7.3`, so probably that's where the methods were removed: https://developer.android.com/reference/com/google/android/play/core/release-notes#1-7-3

If I remember correctly this allowed you to set the kind of update that the app supports, either `FULL` or `FLEXIBLE`

**Comment by adriangl** [comment]
> @adriangl how we could test that the new version still works? using internal testing works good?

Looks like it, [according to the docs](https://developer.android.com/guide/playcore/in-app-updates#internal-app-sharing). 
I honestly don't remember if that option was available when I first implemented the lib (I'd say no), so I can't confirm if it works as expected

File: lib/src/main/java/com/bq/appupdateshelper/FakeAppUpdatesHelper.java (hunk 1 above)
Comment (danielceinos) [reply lib/src/main/java/com/bq/appupdateshelper/FakeAppUpdatesHelper.java:84]: But the kind of update you set it up on `onUpdateStatusResult` with the `requestCode` param. SO maybe this is not necessary anymore

File: lib/build.gradle [hunk 3]
```diff
@@ -1,15 +1,20 @@
...
 
 android {
-    compileSdkVersion project.ext.compileSdkVersion
+    compileSdkVersion 30
 
     defaultConfig {
-        minSdkVersion project.ext.minSdkVersion
-        targetSdkVersion project.ext.targetSdkVersion
-        versionCode 1
+        minSdkVersion 21
```
Comment (danielceinos) [inline lib/build.gradle:14]: What do you think about doesnt couple example app with the lib? @adriangl

Comment (adriangl) [reply lib/build.gradle:14]: You mean the `minSdkVersion`, or the `app`'s  `versionCode`?

If it's the former, you could update the ones in the root's `build.gradle`, which currently point to API 29 and use them here as well as in `app`.

If it's the latter, I'd keep `app`'s `versionCode` to 1 for a very simple reason: so we always have the lowest version code available to test the in-app updates there :P

If you meant anything else, I didn't get it xD

File: lib/src/main/java/com/bq/appupdateshelper/FakeAppUpdatesHelper.java (hunk 1 above)
Comment (adriangl) [reply lib/src/main/java/com/bq/appupdateshelper/FakeAppUpdatesHelper.java:84]: Yeah, it may not be needed, so leave it deleted, it's OK

**Review by adriangl — CHANGES_REQUESTED** [review]

File: lib/build.gradle (hunk 3 above)
Comment (danielceinos) [reply lib/build.gradle:14]: I was talking only about lib gradle file. Its a bit awkward need to look at the example app to get what `minSdk` version the library has, do you understand why I mean?

I'm OK with has `versionCode` to 1 at app's gradle.

Comment (adriangl) [reply lib/build.gradle:14]: Sorry for the delay. Yeah, we can leave the versions decoupled ;)

**Review by adriangl — APPROVED** [review]

**Pull request merged**